packages/puppeteer-core/src/bidi/Frame.ts

```diff
@@ -270,4 +246,47 @@ export class BidiFrame extends Frame {
       throw error;
     }
   }
+
+  /** @internal */
+  async _waitBidiLifecycleAndNetworkIdle(
```

Comment:
I'm all for making another observable, but the beauty of RxJS really comes from explicit pipelines. Could we split this up into smaller operators that get composed rather than one large operator?

Reply:
Let's chat offline about it.